persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/UpdateDescriptionService.java

```diff
@@ -0,0 +1,185 @@
+import edu.unc.lib.dl.persist.services.storage.StorageLocation;
+import edu.unc.lib.dl.persist.services.storage.StorageLocationManager;
+import edu.unc.lib.dl.persist.services.transfer.BinaryTransferService;
+import edu.unc.lib.dl.persist.services.transfer.BinaryTransferSession;
+import edu.unc.lib.dl.services.OperationsMessageSender;
+import edu.unc.lib.dl.validation.MODSValidator;
+import edu.unc.lib.dl.validation.MetadataValidationException;
+import io.dropwizard.metrics5.Timer;
+
+/**
+ * Service that manages description, e.g., MODS, updates
+ *
+ * @author harring
+ *
+ */
+public class UpdateDescriptionService {
+    private static final Logger log = LoggerFactory.getLogger(UpdateDescriptionService.class);
+
+    private AccessControlService aclService;
+    private RepositoryObjectLoader repoObjLoader;
+    private OperationsMessageSender operationsMessageSender;
+    private MODSValidator modsValidator;
+    private BinaryTransferService transferService;
+    private StorageLocationManager locationManager;
+
+    private boolean validate;
+
+    private static final Timer timer = TimerFactory.createTimerForClass(UpdateDescriptionService.class);
+
+    public UpdateDescriptionService() {
+        validate = true;
+    }
+
+    /**
+     * Updates the MODS description of a single object
+     *
+     * @param agent
+     * @param pid
+     * @param modsStream
+     * @throws MetadataValidationException
+     * @throws IOException
+     */
+    public void updateDescription(AgentPrincipals agent, PID pid, InputStream modsStream)
+            throws MetadataValidationException, IOException {
+
+        ContentObject obj = (ContentObject) repoObjLoader.getRepositoryObject(pid);
+        StorageLocation destLocation = locationManager.getStorageLocation(obj);
+
+        try (BinaryTransferSession transferSession = transferService.getSession(destLocation)) {
+            updateDescription(transferSession, agent, obj, modsStream);
+        }
+    }
+
+    /**
+     * Updates the MODS description of an object as part of the provided ongoing session.
+     *
+     * @param transferSession ongoing transfer session
+     * @param agent
+     * @param pid
+     * @param modsStream
+     * @throws MetadataValidationException
+     * @throws IOException
+     */
+    public void updateDescription(BinaryTransferSession transferSession, AgentPrincipals agent,
+            PID pid, InputStream modsStream) throws MetadataValidationException, IOException {
+
+        ContentObject obj = (ContentObject) repoObjLoader.getRepositoryObject(pid);
+
+        updateDescription(transferSession, agent, obj, modsStream);
+    }
+
+    private void updateDescription(BinaryTransferSession transferSession, AgentPrincipals agent,
+            ContentObject obj, InputStream modsStream) throws IOException {
+
+        log.debug("Updating description for {}", obj.getPid().getId());
+        try (Timer.Context context = timer.time()) {
```

Comment:
Does the timer get used?

Reply:
If I remember correctly, when the `try` block ends the timer gets closed which writes the result to the metrics client, which gets dumped to a log periodically